### ISSUE TITLE
Missed bad file modes in file with multiple resource bodies

### ISSUE THREAD

**Issue by paran1** [rodjek/puppet-lint#663]
In a file resource with multiple resource body, both file_mode and unquoted_file_mode miss subsequent bad file modes if there is a selector in one of the resource bodies.

Tested using puppet-lint 2.1.1 and current master (e90c1f4bf73823155268ab893708b371a723f85c)

(The examples below violate the style guide by not having a default body. That was removed to create more minimal test cases.)

### Working example
Both `file_mode` and `unquoted_file_mode` warns about three files as expected.
```puppet
# multi_body_file_bad_modes.pp
file {
  '/tmp/foo1':
    mode => 644;
  '/tmp/foo2':
    mode => 644;
  '/tmp/foo3':
    mode => 644;
}
```
```
$ puppet-lint multi_body_file_bad_modes.pp
WARNING: unquoted file mode on line 4
WARNING: unquoted file mode on line 6
WARNING: unquoted file mode on line 8
WARNING: mode should be represented as a 4 digit octal value or symbolic mode on line 4
WARNING: mode should be represented as a 4 digit octal value or symbolic mode on line 6
WARNING: mode should be represented as a 4 digit octal value or symbolic mode on line 8
```

### Bug example 1
Same code, with a selector added in the first file resource body.
`selector_inside_resource` correctly warns about the selector.
But both file mode checks now only warns about the first file.

```puppet
# multi_body_file_bad_modes_selector.pp
file {
  '/tmp/foo1':
    ensure => $foo ? { default => absent },
    mode   => 644;
  '/tmp/foo2':
    mode => 644;
  '/tmp/foo3':
    mode => 644;
}
```
```
$ puppet-lint multi_body_file_bad_modes_selector.pp
WARNING: selector inside resource block on line 4
WARNING: unquoted file mode on line 5
WARNING: mode should be represented as a 4 digit octal value or symbolic mode on line 5
```
### Bug example 2
Same code, with the first file mode fixed.
The later file modes are still ignored, so this example gives no file mode warnings.
```puppet
# multi_body_file_bad_modes_selector_first_ok.pp
file {
  '/tmp/foo1':
    ensure => $foo ? { default => absent },
    mode   => '0644';
  '/tmp/foo2':
    mode => 644;
  '/tmp/foo3':
    mode => 644;
}
```
```
$ puppet-lint multi_body_file_bad_modes_selector_first_ok.pp
WARNING: selector inside resource block on line 4
```

**Comment by rothsa** [comment]
I think this is because the lexer is seeing the ? as a resource rather than a control flow operator. 

Each of the modes that it doesn't recognize has their resource[:type].value as showing as '?' rather than 'file'. The subsequent modes become param_tokens of '?' as a result instead of param_tokens of 'file'.

I haven't figured out how exactly to except :QMARK but pretty sure it would be somewhere in this method: https://github.com/rodjek/puppet-lint/blob/master/lib/puppet-lint/data.rb#L107

**Comment by paran1** [comment]
I ended up having some free time to spend on this today.
I believe #666 fixes the code to avoid matching the :QMARK. Thanks @rothsa for finding out what to look for.

**Comment by rothsa** [comment]
Sure! Here are the tests I used if they are worth including, based on your description: https://github.com/rodjek/puppet-lint/pull/667

**Comment by rothsa** [comment]
They are not according to appveyor, WOMM :/ Nevermind then.

**Comment by paran1** [comment]
Isn't that because the tests are failing when you don't have the fix?

**Comment by rothsa** [comment]
Yes 🤦‍♀️  Made a pull on your pull.
